src/rpc/net.cpp

```diff
@@ -156,7 +156,10 @@ static UniValue getpeerinfo(const JSONRPCRequest& request)
     for (const CNodeStats& stats : vstats) {
         UniValue obj(UniValue::VOBJ);
         CNodeStateStats statestats;
-        bool fStateStats = GetNodeStateStats(stats.nodeid, statestats);
+        bool fStateStats{false};
+        if (node.peerman) {
```

Comment:
It is currently not possible to not have `node.peerman`. I think an `Ensure..()` like `EnsureMempool` would be good. It doesn't really make sense to call `getpeerinfo` without a peerman anyway?

Reply:
I've done something slightly different and checked existence at the top of this function. Let me know what you think.